packages/ui-components/src/components/Message/Message.test.tsx

```diff
@@ -0,0 +1,162 @@
+  })
+
+  test("renders text from children, if provided", () => {
+    render(<Message data-testid="my-message">{"My Message children text goes here!"}</Message>)
+    expect(screen.getByTestId("my-message")).toHaveTextContent("My Message children text goes here!")
+  })
+
+  test("prioritizes children over 'text' prop when both are provided", () => {
+    render(
+      <Message data-testid="my-message" text="I should not be here.">
+        {"My Message children text goes here!"}
+      </Message>
+    )
+    expect(screen.getByTestId("my-message")).toHaveTextContent("My Message children text goes here!")
+  })
+
+  test("applies custom classNames, if provided", () => {
+    render(<Message data-testid="my-message" className="my-custom-class" />)
+    expect(screen.getByTestId("my-message")).toHaveClass("my-custom-class")
+  })
+
+  test("renders the component with all provided props", () => {
+    render(<Message data-testid="my-message" name="My shiny little Message" />)
+    expect(screen.getByTestId("my-message")).toHaveAttribute("name", "My shiny little Message")
+  })
+
+  // VARIANTS
+
+  test("renders an info Message by default if no variant is provided", () => {
+    render(<Message data-testid="my-message" />)
+    expect(screen.getByTestId("my-message")).toHaveClass("juno-message-info")
+    expect(screen.getByRole("img")).toHaveClass("jn-text-theme-info")
+  })
+
+  test.each`
+    variant      | cssClass                  | iconClass
+    ${"info"}    | ${"juno-message-info"}    | ${"jn-text-theme-info"}
+    ${"warning"} | ${"juno-message-warning"} | ${"jn-text-theme-warning"}
+    ${"danger"}  | ${"juno-message-danger"}  | ${"jn-text-theme-danger"}
+    ${"error"}   | ${"juno-message-error"}   | ${"jn-text-theme-error"}
+    ${"success"} | ${"juno-message-success"} | ${"jn-text-theme-success"}
+  `(
+    "renders a $variant Message with the correct classes",
+    ({ variant, cssClass, iconClass }: { variant: VariantType; cssClass: string; iconClass: string }) => {
+      render(<Message data-testid="my-message" variant={variant} />)
+      expect(screen.getByTestId("my-message")).toHaveClass(cssClass)
+      expect(screen.getByRole("img")).toHaveClass(iconClass)
+    }
+  )
+
+  test("ensures the icon cannot shrink", () => {
+    render(<Message data-testid="my-message" />)
+    expect(screen.getByTestId("my-message")).toHaveClass("juno-message-info")
+    expect(screen.getByRole("img")).toHaveClass("jn-shrink-0")
+  })
+
+  // DISMISSIBLE BEHAVIOUR
+
+  test("renders a Message without a dismiss button by default", () => {
+    render(<Message data-testid="my-message" />)
+    expect(screen.queryByRole("button", { name: /close/i })).not.toBeInTheDocument()
+  })
+
+  test("renders a dismissible Message when 'dismissible' is true", async () => {
+    render(<Message data-testid="my-message" dismissible={true} />)
+
+    const user = userEvent.setup()
```

Comment:
Do we need a setup here?

Reply:
We probably don't need it for such a simple test, though it's recommended for more reliable and accurate test results. It helps handle asynchronous state updates and side effects. But just fyi: We use it across our code base, I didn't add it here. 🤪